### PR TITLE
CUMULUS-1338: Update sfSemaphoreDown lambda to be triggered via Cloudwatch events

### PR DESCRIPTION
**Pull request by markdboyd**
**Summary:** Summary of changes

Addresses [CUMULUS-1338: Update sfSemaphoreDown lambda to be triggered via Cloudwatch](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1338)

## Changes

* Updated sfSemaphoreDown lambda to be triggered using Cloudwatch rule for Step Function updates
* Updated unit tests for sfSemaphoreDown lambda

## PR Checklist

- [ ] Update CHANGELOG
- [x] Unit tests

### CODE REVIEW

File: packages/api/lambdas/sf-semaphore-down.js [hunk 1]
```diff
@@ -12,14 +10,25 @@ const {
...
 /**
  * Determine if workflow is in a terminal state.
  *
- * @param {Object} message - A workflow message object
+ * @param {Object} status - A Step Function execution status
  * @returns {boolean} - True if workflow is in terminal state.
  */
-const isTerminalMessage = (message) =>
-  message.meta.status === 'failed' || message.meta.status === 'completed';
+const isTerminalMessage = (status) =>
```
Comment (yjpa7145) [inline packages/api/lambdas/sf-semaphore-down.js]:
```suggestion
const isTerminalStatus = (status) =>
```

File: packages/api/lambdas/sf-semaphore-down.js [hunk 2]
```diff
@@ -12,14 +10,25 @@ const {
...
+  status === 'FAILED'
```
Comment (yjpa7145) [inline packages/api/lambdas/sf-semaphore-down.js]: There are a lot of cases where we do a check like this. I think we may want to add an `isOneOf()` function to `@cumulus/common/utils`

Comment (markdboyd) [reply packages/api/lambdas/sf-semaphore-down.js]: Would that be better than just doing `['FAILED', 'COMPLETED', 'TIMED_OUT', 'ABORTED'].includes(status)`?

File: packages/api/lambdas/sf-semaphore-down.js [hunk 3]
```diff
@@ -62,34 +71,39 @@ async function decrementQueueSemaphore(queueName) {
...
-function getSemaphoreDecrementTasks(event) {
-  return get(event, 'Records', [])
-    // Skip if this record is not from SNS or if the SNS message is empty
-    .filter((record) => has(record, 'Sns.Message') && !isNil(record.Sns.Message))
-    .map((record) => JSON.parse(record.Sns.Message))
-    .filter((message) => isDecrementMessage(message))
-    .map((message) => getQueueName(message))
-    .map((queueName) => decrementQueueSemaphore(queueName));
+async function handleSemaphoreDecrementTask(event) {
+  if (!isSfExecutionEvent(event)) {
```
Comment (markdboyd) [inline packages/api/lambdas/sf-semaphore-down.js]: I don't love have two separate early returns (one here and below), but it seemed unnecessary to proceed further if the event is not for a Step Function. 

I separately don't love having to do `return Promise.resolve`, but ESlint requires it

File: packages/api/lambdas/sf-semaphore-down.js [hunk 4]
```diff
@@ -62,34 +71,39 @@ async function decrementQueueSemaphore(queueName) {
...
+    return Promise.resolve();
+  }
+
+  const status = get(event, 'detail.status');
+  const output = get(event, 'detail.output', '{}');
+  const message = JSON.parse(output);
```
Comment (markdboyd) [inline packages/api/lambdas/sf-semaphore-down.js]: It seems possible that `output` could just be a regular string or some other value that is not a JSON string. I think this needs to be more defensive

File: packages/api/lambdas/sf-semaphore-down.js (hunk 3 above)
Comment (yjpa7145) [reply packages/api/lambdas/sf-semaphore-down.js]: This function doesn't need to return anything, so you could just say `if (...) return` and then `await decrementQueueSemaphore()` at the end.

File: packages/api/lambdas/sf-semaphore-down.js (hunk 2 above)
Comment (yjpa7145) [reply packages/api/lambdas/sf-semaphore-down.js]: I think it makes it more readable.

**Pull request merged**